File: internal/indexer/testdata/java/services/CoreModelService.java
package services;

import models.CoreModel;
import java.util.ArrayList;
import java.util.List;

/** CoreModelService manages CoreModel instances with high-coupling patterns.
 * Demonstrates: import (0.40), extends (1.0 via lexical), instantiation (0.95), method calls (0.65). */
public class CoreModelService extends CoreModel {

    private List<CoreModel> store = new ArrayList<>();

    /** Creates a CoreModelService seeded with an initial CoreModel. */
    public CoreModelService() {
        super("service-root", 1.0);
        CoreModel initial = new CoreModel("initial", 0.5);
        store.add(initial);
    }

    /** save stores a CoreModel entry (method call coupling). */
    public boolean save(CoreModel model) {
        if (model.isValid()) {
            store.add(model);
            return true;
        }
        return false;
    }

    /** findByTitle looks up a CoreModel by its title. */
    public CoreModel findByTitle(String title) {
        for (CoreModel m : store) {
            if (title.equals(m.getTitle())) {
                return m;
            }
        }
        return null;
    }

    /** findAll returns every stored CoreModel. */
    public List<CoreModel> findAll() {
        return store;
    }

    /** create instantiates a fresh CoreModel (instantiation coupling). */
    public static CoreModel create(String title, double score) {
        return new CoreModel(title, score);
    }
}

File: internal/indexer/testdata/ts/services/coreModelService.ts
import { CoreModel } from "../models/coreModel";

/** CoreModelService extends CoreModel and manages a collection of instances.
 * Demonstrates: import (0.40), extends (1.0 via lexical), instantiation new (0.95),
 * method calls (0.65), and type annotations (0.25). */
export class CoreModelService extends CoreModel {
  private store: CoreModel[] = [];

  constructor() {
    super("service-root", 1.0);
    const initial = new CoreModel("initial", 0.5);
    this.store.push(initial);
  }

  /** save stores a CoreModel entry. */
  save(model: CoreModel): boolean {
    if (model.isValid()) {
      this.store.push(model);
      return true;
    }
    return false;
  }

  /** findByTitle looks up a CoreModel by its title. */
  findByTitle(title: string): CoreModel | undefined {
    return this.store.find((m) => m.title === title);
  }

  /** findAll returns every stored CoreModel. */
  findAll(): CoreModel[] {
    return this.store;
  }

  /** create instantiates a fresh CoreModel. */
  static create(title: string, score: number): CoreModel {
    return new CoreModel(title, score);
  }
}

/** storeCoreModel persists a CoreModel instance. */
export function storeCoreModel(model: CoreModel): boolean {
  return model.isValid();
}
